File: Codigos estudiantes por lenguaje/C++/Cristian Maldonado/Geometria/Arranged_Coin_Triangle.cpp
#include <iostream>

using namespace std;

float squareRoot(float n){
    float x = n;
    float y = 1;
    float e = 0.000001f;
    while((x - y) > e){
        x = (x + y)/2;
        y = n / x;
    }
    return x;
}

int findMaximumHeight(int N){
    int n = 1 + 8 * N;
    int maxH = (int)(-1 + squareRoot(n))>>1;
    return maxH;
}

int main()
{
    ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
    int N; cin>>N;
    cout<<findMaximumHeight(N)<<endl;
    return 0;
}

File: Codigos estudiantes por lenguaje/Java/Cristian Maldonado/Geometria/Arranged_Coin_Triangle.java

public class Arranged_Coin_Triangle {

    static float squareRoot(float n) {
        float x = n;
        float y = 1;
        float e = 0.000001f;
        while (x - y > e) {
            x = (x + y) / 2;
            y = n / x;
        }
        return x;
    }

    static int findMaximumHeight(int N) {
        int n = 1 + 8 * N;
        int maxH = (int) (-1 + squareRoot(n)) / 2;
        return maxH;
    }

    public static void main(String[] args) {
        int N = 12;
        System.out.println(findMaximumHeight(N));
    }
}
